com.stansassets.build/Editor/Core/BuildExecutor.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace StansAssets.Build.Editor
+{    
+    /// <summary>
+    /// Run registered steps and tasks with build on result
+    /// </summary>
+    public static class BuildExecutor
+    {
+        private static List<IBuildStep> s_Steps = new List<IBuildStep>();
+        private static List<IBuildTask> s_Task = new List<IBuildTask>();
+        
+        /// <summary>
+        /// Add IBuildStep object to build pipeline as a step
+        /// </summary>
+        /// <param name="step">Build step</param>
+        public static void RegisterStep(IBuildStep step)
+        {
+            s_Steps.Add(step);
+        }
+        
+        /// <summary>
+        /// Add IBuildTask object to build pipeline as a task
+        /// </summary>
+        /// <param name="buildTask">Build task</param>
+        public static void RegisterScenePostprocessTask(IBuildTask buildTask)
+        {
+            s_Task.Add(buildTask);
+        }
+        
+        /// <summary>
+        /// Run build process with included steps and tasks
+        /// </summary>
+        public static void Build(BuildContext buildContext)
+        {
+            RegisterUnityPlayerBuildStep();
+            
+            SortSteps();
+            SortTasks();
+            
+            RunSteps(buildContext);
+            RunTasks(buildContext);
```

Comment:
feel like we will still run tasks if steps are failed

Reply:
Another thing whet we should run scene post-processing tasks?
It seems like it's to late to do something when all the build sets were completed.
i.e. `RegisterUnityPlayerBuildStep ` is also a step and once it's started we can't change anything anymore since we will go into `post-processing mode`

Reply:
I've swapped 42 and 43 rows
I will add process abort logic based on @OlegTutevych future commit